CaC Runbooks blog (#1196)
OctopusDeploy/blog: blog/2025-03/config-as-code-runbooks/index.md

isFeatured: false
tags: 
  - Product
  - Configuration as Code
  - Runbooks
---

Config as Code for Runbooks is officially landing as part of Octopus 2025.1. This feature gives you a seamless, version-controlled way to manage your runbooks. 

You can run, create, and track your runbooks in your chosen Git repository, bringing your runbooks together with your deployment process. This means fewer places to make changes and provides more control over your workflows.

## Why Config as Code for Runbooks?

[Octopus Runbooks] (https://octopus.com/docs/runbooks) automates routine and emergency operations tasks. Storing your runbooks in your Git repo with your application code, deployment process, and non-sensitive variables unlocks the full power of version control.

- Store your runbooks in your Git repo to stay in sync. You can make and test changes alongside your app code, so your code and operational processes never get out of sync.
- Store your runbooks in Git to version-control your operational tasks. This way, you can manage versions with minimal effort. If you need to roll back or retrieve an earlier configuration, you can run the runbook for a specific commit or tag.

> - Access the complete history for audibility. You can review all changes via pull requests, providing a full history of all changes to your runbooks.

Reviewer: ```suggestion
- Access the complete history for auditability. You can review all changes via pull requests, providing a full history of all changes to your runbooks.
```

- If your projects are version-controlled, the setup and migration is quick and easy. You'll also find using Config as Code for Runbooks natural. Following Git principles will make the process straightforward.
- If you're not using version control yet, you can enable it and start using Config as Code for Runbooks. [See our docs for a guide on enabling version control for your project](https://octopus.com/docs/projects/version-control/converting).

### Migrating existing runbooks

If you use Octopus Runbooks, we have a tool to help you migrate to Config as Code for Runbooks.

- Migration process: When you start the migration, Octopus moves your published runbooks to a new folder, `/runbooks`, in your chosen repository. This folder stores the published snapshot and configuration from the runbook.
- Octopus migrates any draft versions of your runbook to a separate sub-folder `/runbooks/migrated-draft`. You won't see these in the UI, but you can access them by moving them into the main runbooks folder using a pull request (PR).
- History retention: You don't need to worry about losing your history. The UI will still show your existing runbook runs.

> ![Screen recording of Runbooks migration process. Starting on the Runbook list, the user opens the wizard, reviews the changes, creates a new branch named feat/migrate-runbooks, uses the default commit message, reviews the migration and completes it. Once completed, the Runbooks list screen is automatically refreshed showing a branch selector and an alert showing there are un-migrated drafts](configascode.gif) 

Reviewer: ```suggestion
![GIF of Runbooks migration process. Starting on the Runbook list, user opens the wizard, reviews changes, creates new branch, uses default commit message, reviews the migration, and completes it. Next, the Runbooks list screen automatically refreshes showing a branch selector and alert with unmigrated drafts.](configascode.gif) 
```
